bash.sh

```diff
@@ -12,40 +12,83 @@
 #
 # The only proper way to do this is by looping through each item. Other
 # solutions claim to do this using string matching, but they are often
-# dangerous because don't handle edge cases.
+# dangerous because they don't handle edge cases, e.g. may be susceptible to
+# partial match or may not properly handle elements containing newlines.
 #
 
 #
 # Returns 0 if item is in the array; 1 otherwise.
 #
-# $1: The array value to search for
-# $@: The array values, e.g. "${myarray[@]}"
+# $1: name of the array variable to search
+# $2: array item to search for
 #
 array_contains() {
-    local item=$1; shift
-    for val in "$@"; do
+    local arr_name=$1
+    local item=$2
+
+    eval "local -a _tmp_arr=(\"\${${arr_name}[@]}\")"
```

Comment:
Sorry, but I think this should be explained in the comment.

Maybe a separate section, then a point to that explanation in each of the functions that uses it?
(yeah, I would repeat the comment in every test, I think)